sdv/data_processing/data_processor.py

```diff
@@ -572,8 +572,7 @@ def _create_config(self, data, columns_created_by_constraints):
                         column, sdtype, column_metadata, is_numeric
                     )
                     sdtypes[column] = 'text'
-
-                else:
+                elif column in self._keys or column_metadata.get('pii'):
```

Comment:
I don't think the metadata should ever have the 'pii' field if the sdtype is an id. The accepted args for id are
https://github.com/sdv-dev/SDV/blob/6d7d67e43d2d6e56a6e5df686573ec0cd479bf52/sdv/metadata/single_table.py#L43-L50
we raise an error otherwise
https://github.com/sdv-dev/SDV/blob/6d7d67e43d2d6e56a6e5df686573ec0cd479bf52/sdv/metadata/single_table.py#L224